src/main/scala/vectorpipe/osm/package.scala

```diff
@@ -27,48 +27,125 @@ package object osm {
   /** Given a path to an OSM XML file, parse it into usable types. */
   def fromLocalXML(path: String)(implicit sc: SparkContext): Either[String, (RDD[Node], RDD[Way], RDD[Relation])] = {
     /* A byte stream, so as to not tax the heap */
-    Try(new FileInputStream(path) : InputStream).flatMap(xml => Element.elements.parse(xml)) match {
+    Try(new FileInputStream(path): InputStream).flatMap(xml => Element.elements.parse(xml)) match {
       case Failure(e) => Left(e.toString)
       case Success((ns, ws, rs)) =>
         Right((sc.parallelize(ns), sc.parallelize(ws), sc.parallelize(rs)))
     }
   }
 
-  /** Convert an RDD of raw OSM [[Element]]s into interpreted GeoTrellis
-    * [[Feature]]s. In order to mix the various subtypes together, they've
-    * been upcasted internally to [[Geometry]]. Note:
-    * {{{
-    * type OSMFeature = Feature[Geometry, Tree[ElementData]]
-    * }}}
-    *
-    * ===Behaviour===
-    * This algorithm aims to losslessly "sanitize" its input data,
-    * in that it will break down malformed Relation structures, as
-    * well as cull member references to Elements which no longer
-    * exist (or exist outside the subset of data you're working
-    * on). Mathematically speaking, there should exist a function
-    * to reverse this conversion. This theoretical function and
-    * `toFeatures` form an isomorphism if the source data is
-    * correct. In other words, given:
-    * {{{
-    * parse: XML => RDD[Element]
-    * toFeatures: RDD[Element] => RDD[OSMFeature]
-    * restore: RDD[OSMFeature] => RDD[Element]
-    * unparse: RDD[Element] => XML
-    * }}}
-    * then:
-    * {{{
-    * unparse(restore(toFeatures(parse(xml: XML))))
-    * }}}
-    * will yield a body of semantically correct OSM data.
-    *
-    * To achieve this sanity, the algorithm has the following behaviour:
-    *   - Graphs of [[Relation]]s will be broken into spanning [[Tree]]s.
-    *   - It doesn't make sense to represent non-multipolygon Relations as
-    *     GeoTrellis `Geometry`s, so Relation metadata is disseminated
-    *     across its child members. Otherwise, Relations are "dropped"
-    *     from the output.
+  /** Given a path to an Apache ORC file containing OSM data, read out RDDs of each Element type.
+    * If you want to read a file from S3, you must call [[vectorpipe.useS3]] first
+    * to properly configure Hadoop to read your S3 credentials.
     */
+  def fromORC(path: String)(implicit hc: HiveContext): Either[String, (RDD[Node], RDD[Way], RDD[Relation])] = {
```

Comment:
Maybe `HiveContext` should be created *inside* this function, so that the user doesn't have to worry about it at all. It would save them an explicit dependency and an import in their main file.